airbyte-cdk/python/unit_tests/sources/declarative/test_manifest_declarative_source.py

```diff
@@ -727,3 +742,419 @@ def test_given_debug_when_read_then_set_log_level(self, declarative_source_read)
+                         "request_headers": {},
+                         "request_body_json": {},
+                         "authenticator": {
+                             "type": "ApiKeyAuthenticator",
+                             "header": "apikey",
+                             "api_token": "{{ config['api_key'] }}"
+                         }
+                     },
+                     "partition_router": {
+                         "type": "ListPartitionRouter",
+                         "values": ["0", "1"],
+                         "cursor_field": "partition"
+                     },
+                     "record_selector": {
+                         "type": "RecordSelector",
+                         "extractor": {
+                             "type": "DpathExtractor",
+                             "field_path": [
+                                 "rates"
+                             ]
+                         }
+                     },
+                     "paginator": {
+                         "type": "DefaultPaginator",
+                         "page_size": 2,
+                         "page_size_option": {"inject_into": "request_parameter", "field_name": "page_size"},
+                         "page_token_option": {"inject_into": "path", "type": "RequestPath"},
+                         "pagination_strategy": {"type": "CursorPagination", "cursor_value": "{{ response._metadata.next }}", "page_size": 2},
+                     },
+                 }
+             }
+         ],
+         "spec": {
+             "connection_specification": {
+                 "$schema": "http://json-schema.org/draft-07/schema#",
+                 "type": "object",
+                 "required": [
+                     "api_key"
+                 ],
+                 "properties": {
+                     "api_key": {
+                         "type": "string",
+                         "title": "API Key",
+                         "airbyte_secret": True
+                     }
+                 },
+                 "additionalProperties": True
+             },
+             "documentation_url": "https://example.org",
+             "type": "Spec"
+         }
+     },
+     (
+             _create_page({"rates": [{"ABC": 0, "partition": 0}, {"AED": 1, "partition": 0}], "_metadata": {"next": "next"}}),
+             _create_page({"rates": [{"USD": 3, "partition": 0}], "_metadata": {}}),
+             _create_page({"rates": [{"ABC": 2, "partition": 1}], "_metadata": {}}),
+     ),
+     [{"ABC": 0, "partition": 0}, {"AED": 1, "partition": 0}, {"USD": 3, "partition": 0}, {"ABC": 2, "partition": 1}],
+     [call({"partition": "0"}, {}, None), call({"partition": "0"}, {}, {"next_page_token": "next"}), call({"partition": "1"}, {}, None),]
+     )
+])
+def test_read_manifest_declarative_source(test_name, manifest, pages, expected_records, expected_calls):
+    _stream_name = "Rates"
+    config = {"__injected_declarative_manifest": manifest}
```

Comment:
why do we need to inject the manifest into the config? Can't we just pass the manifest as its own parameter to `_run_read()` and then skip calling `config.get("__injected_declarative_manifest")`. 

Reading this gave me the impression we're testing the `connector_builder` entrypoint because of the way we inject it into the config which is slightly misleading since this is very much just the common flow for all low-code processing